src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,179 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> head;
+    private Node<T> tail;
+
     @Override
     public boolean add(T value) {
-        return false;
+        if (head == null) {
+            head = new Node<>(null, value, null);
+            tail = new Node<>(null, value, null);
+        } else {
+            if (head.next == null) {
```

Comment:
try to simplify this if else. you can check for size == 0 and assign head and tail due to situation